code/modules/projectiles/gun.dm

```diff
@@ -175,7 +175,24 @@
 		user << "<span class='userdanger'>You need both hands free to fire \the [src]!</span>"
 		return
 
-	process_fire(target,user,1,params)
+	//DUAL WIELDING
+	var/bonus_spread = 0
+	var/obj/item/weapon/gun/off_hand
+	if(ishuman(user) && user.a_intent == "harm")
+		var/mob/living/carbon/human/H = user
+		if(H.r_hand == src && istype(H.l_hand, /obj/item/weapon/gun))
+			off_hand = H.l_hand
+
+		else if(H.l_hand == src && istype(H.r_hand, /obj/item/weapon/gun))
+			off_hand = H.r_hand
+
+	if(off_hand && off_hand.can_trigger_gun(user))
+		bonus_spread = 12 * off_hand.weapon_weight
+		spawn(1)
```

Comment:
`addtimer` would have a delay of 2 deciseconds here because it only ticks once every 2 deciseconds...though it *can* be used in place of `spawn(0)`...but that has a lot more overhead, comparatively (I don't understand the `addtimer(0)` obsession).